### PR TITLE
[systemtest] Refactor of resource classes and STs - prerequisite for fabric8 upgrade

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Refactoring

### Description

This PR is small prerequisite before the fabric8 upgrade and (maybe) for the paralelism PRs from @see-quick.
Because in new version of fabric8 they are removing the Doneable classes, we had to rework our resource classes and STs (as we used it everywhere). So (after discussion with @Frawless and little look to @see-quick 's code) I decided to have the `create(..)` method which will create resource from the builders.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java [hunk 1]
```diff
@@ -109,13 +109,13 @@ void testMirrorMaker2() {
...
 
         // Deploy source kafka
-        KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 1, 1).done();
+        KafkaResource.create(KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 1, 1).build());
         // Deploy target kafka
-        KafkaResource.kafkaEphemeral(kafkaClusterTargetName, 1, 1).done();
+        KafkaResource.create(KafkaResource.kafkaEphemeral(kafkaClusterTargetName, 1, 1).build());
         // Deploy Topic
-        KafkaTopicResource.topic(kafkaClusterSourceName, topicSourceName, 3).done();
+        KafkaTopicResource.topic(kafkaClusterSourceName, topicSourceName, 3).build();
```
Comment (sknot-rh) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java:116]:
```suggestion
        KafkaTopicResource.create(KafkaTopicResource.topic(kafkaClusterSourceName, topicSourceName, 3).build());
```

**Review by sknot-rh — CHANGES_REQUESTED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java [hunk 2]
```diff
@@ -209,7 +209,7 @@ void testProducerConsumerMirrorMaker() {
                     .withOAuthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
                     .build();
 
-                kafkaOauthClientJob.consumerStrimziOauthPlain().done();
+                kafkaOauthClientJob.create(kafkaOauthClientJob.consumerStrimziOauthPlain().build());
```
[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java:212]: As we are doing for the producer -> I just added deletion of the job. Good catch!

[the remark this answers is not in the document]
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java:212]: 1) I added removal of the job (as we have it with producer)
2) the job here wasn't never "deployed" as I'm looking to the history -> it kept the "successful" stage from previous consumer job (which wasn't deleted), now when I'm trying to send messages, the job errors with TLS handshake exception. Just FYI @Frawless 

I'm gonna fix it ASAP

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Review by sknot-rh — APPROVED** [review]
LGTM

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests true profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]
Just make sure that all tests are working, even upgrade tests.

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=io.strimzi.systemtest.security.oauth.OauthTlsST#testProducerConsumerBridge,io.strimzi.systemtest.security.OpaIntegrationST#testOpaAuthorization,io.strimzi.systemtest.bridge.HttpBridgeKafkaExternalListenersST#testTlsAuthWithWeirdUsername,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaAdminClient,io.strimzi.systemtest.operators.topic.TopicST#testCreateDeleteCreate,io.strimzi.systemtest.operators.topic.TopicST#testTopicModificationOfReplicationFactor,io.strimzi.systemtest.operators.topic.TopicST#testSendingMessagesToNonExistingTopic,io.strimzi.systemtest.operators.topic.TopicST#testMoreReplicasThanAvailableBrokers,io.strimzi.systemtest.operators.topic.TopicST#testCreateTopicViaKafka,io.strimzi.systemtest.log.LoggingChangeST#testJSONFormatLogging,io.strimzi.systemtest.log.LoggingChangeST#testDynamicallySetEOloggingLevels,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testCapacityFile,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testDeployAndUnDeployCruiseControl,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationDiskChangeDoNotTriggersRollingUpdateOfKafkaPods,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationReflection,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationFileIsCreated,io.strimzi.systemtest.cruisecontrol.CruiseControlConfigurationST#testConfigurationPerformanceOptions,io.strimzi.systemtest.specific.SpecificST#testRackAwareConnectCorrectDeployment,io.strimzi.systemtest.watcher.AllNamespaceST#testUserInDifferentNamespace,io.strimzi.systemtest.tracing.TracingST#testKafkaBridgeService

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=upgrade testcase=StrimziUpgradeST

**Pull request merged**